presto-native-execution/presto_cpp/main/types/PrestoToVeloxConnector.cpp

```diff
@@ -0,0 +1,1388 @@
+  }
+
+  return std::make_unique<common::BigintMultiRange>(
+      std::move(bigintFilters), nullAllowed);
+}
+
+std::unique_ptr<common::Filter> combineBytesRanges(
+    std::vector<std::unique_ptr<common::BytesRange>>& bytesFilters,
+    bool nullAllowed) {
+  bool allSingleValue = std::all_of(
+      bytesFilters.begin(), bytesFilters.end(), [](const auto& range) {
+        return range->isSingleValue();
+      });
+
+  if (allSingleValue) {
+    std::vector<std::string> values;
+    values.reserve(bytesFilters.size());
+    for (const auto& filter : bytesFilters) {
+      values.emplace_back(filter->lower());
+    }
+    return std::make_unique<common::BytesValues>(values, nullAllowed);
+  }
+
+  int lowerUnbounded = 0, upperUnbounded = 0;
+  bool allExclusive = std::all_of(
+      bytesFilters.begin(), bytesFilters.end(), [](const auto& range) {
+        return range->lowerExclusive() && range->upperExclusive();
+      });
+  if (allExclusive) {
+    folly::F14FastSet<std::string> unmatched;
+    std::vector<std::string> rejectedValues;
+    rejectedValues.reserve(bytesFilters.size());
+    for (int i = 0; i < bytesFilters.size(); ++i) {
+      if (bytesFilters[i]->isLowerUnbounded()) {
+        ++lowerUnbounded;
+      } else {
+        if (unmatched.contains(bytesFilters[i]->lower())) {
+          unmatched.erase(bytesFilters[i]->lower());
+          rejectedValues.emplace_back(bytesFilters[i]->lower());
+        } else {
+          unmatched.insert(bytesFilters[i]->lower());
+        }
+      }
+      if (bytesFilters[i]->isUpperUnbounded()) {
+        ++upperUnbounded;
+      } else {
+        if (unmatched.contains(bytesFilters[i]->upper())) {
+          unmatched.erase(bytesFilters[i]->upper());
+          rejectedValues.emplace_back(bytesFilters[i]->upper());
+        } else {
+          unmatched.insert(bytesFilters[i]->upper());
+        }
+      }
+    }
+
+    if (lowerUnbounded == 1 && upperUnbounded == 1 && unmatched.size() == 0) {
+      return std::make_unique<common::NegatedBytesValues>(
+          rejectedValues, nullAllowed);
+    }
+  }
+
+  if (bytesFilters.size() == 2 && bytesFilters[0]->isLowerUnbounded() &&
+      bytesFilters[1]->isUpperUnbounded()) {
+    // create a negated bytes range instead
+    return std::make_unique<common::NegatedBytesRange>(
+        bytesFilters[0]->upper(),
+        false,
+        !bytesFilters[0]->upperExclusive(),
+        bytesFilters[1]->lower(),
+        false,
+        !bytesFilters[1]->lowerExclusive(),
+        nullAllowed);
+  }
+
+  std::vector<std::unique_ptr<common::Filter>> bytesGeneric;
+  for (int i = 0; i < bytesFilters.size(); ++i) {
+    bytesGeneric.emplace_back(std::unique_ptr<common::Filter>(
+        dynamic_cast<common::Filter*>(bytesFilters[i].release())));
+  }
+
+  return std::make_unique<common::MultiRange>(
+      std::move(bytesGeneric), nullAllowed, false);
+}
+
+std::unique_ptr<common::Filter> toFilter(
```

Comment:
Maybe we could move the TupleDomainFilter conversion to a separate class. Though I too am a bit on the fence about it since its used only during SplitConversion.

Reply:
It might be easier to decide where to move once we have more call sites. Let's keep it here for now.